src/data_structures/qgram_index.rs

```diff
@@ -197,7 +197,21 @@ impl QGramIndex {
 
         matches
     }
-}
+
+    /// Return all exact matches of length q
+    /// Complexity O(m + k) for pattern of length m and k being the number of matching q-grams.
+    pub fn q_matches(&self, pattern: &[u8]) -> Vec<(usize, usize)> {
```

Comment:
My feeling is that at the moment, I'd like this to be rather in your new module. I am unsure if this is of general use, especially when needing to allocate a new Vec.